src/lang/lang_types.ml

```diff
@@ -73,7 +73,7 @@ let print_pos ?(prefix = "at ") (start, stop) =
 (** Ground types *)
 
 type ground = ..
-type ground += Bool | Int | String | Float
+type ground += Bool | Int | String | Float | Request
```

Comment:
Nice. I can see why you would want to add `Source` here.

Reply:
You cannot add `Source` here because it depends on the number of channels, so it has to remain a constructed type. I could do this with requests because, as explained above, they don't depend on other types anymore.

Reply:
I meant to say: why you would want to also remove content types from source and then add it here :-)

Reply:
Ah sorry! Let's keep this for the future though...